core/spec/lib/mno_enterprise/smtp_client_spec.rb

```diff
@@ -2,54 +2,93 @@
 
 describe MnoEnterprise::SmtpClient do
   let(:client) { described_class.send(:new) }
+  let(:from_object) { { name: 'John', email: 'john.doe@example.com' } }
```

Comment:
I'm not sure you need to extract all params to their own variable (also `from_object` is not the best name ;) )

```ruby
let(:opts) { nil }
let(:params) do
  [
    'reset-password-instructions',
    {name: 'John', email: 'john.doe@example.com'},	
    {name: 'Joe', email: 'joe.blogg@example.com'},
    attachments: [	
      {	
        name: 'some-file.pdf',	 
        content: 'the file'
      }
    ],
   opts	 
  ] 
end

subject { client.deliver(*params) }
```

Then your context can be:
```ruby
context 'with a specified subject' do
  let(:opts) { subject: 'user specified subject' }
  it 'uses the specified subject' do
    expected_params[:subject] = user_specified_subject
    ...
  end
end
```

You don't need to explicitly test `it 'sets the subject of the email to the #humanize(template)'` as it's tested by `calls mail with the correct params`

Then you don't need to add all the extra context as it's the default ('opts has not been inputted by user').
Also note the context wording as recommended by betterspecs (with/when).